### PR TITLE
Remove lodash.throttle and replace underscore calls with lodash

### PR DESCRIPTION
**Pull request by kristw**
- Remove `lodash.throttle` from dependency since there is now `lodash`.
- Add `babel-plugin-lodash` that helps optimize bundle output by taking only necessary part from lodash instead of the entire bundle.
https://github.com/lodash/babel-plugin-lodash
- Replace `underscore` calls with `lodash` where applicable. 

@williaster @xtinec @conglei

### CODE REVIEW

File: superset/assets/spec/javascripts/modules/utils_spec.jsx [hunk 1]
```diff
@@ -10,12 +9,6 @@ import {
 } from '../../../src/modules/utils';
 
 describe('utils', () => {
-  it('slugify slugifies', () => {
```
Comment (graceguo-supercat) [inline superset/assets/spec/javascripts/modules/utils_spec.jsx]: can we still rewrite but keep this test case?

Comment (kristw) [reply superset/assets/spec/javascripts/modules/utils_spec.jsx]: I tested stubbing Superset's `slugify` implementation with `kebabCase` from `lodash` and it passes all test cases under this `it` except one. 

`kebabCase`: `"camelCase"` => `"camel-case"`
`slugify`: `"camelCase"` => `"camelcase"`

This was the only difference. IMO, I am leaning towards switching to `lodash`'s `kebabCase` which is well-tested library and industry standard than custom implementation and then remove `slugify` from codebase.

I checked where the function `slugify` is being used and it is used for creating tooltip `id` string. My understanding is this is harmless change, so I made the move for it. Please correct me if I misunderstood.

Comment (graceguo-supercat) [reply superset/assets/spec/javascripts/modules/utils_spec.jsx]: it used in creating customized url. for example `/superset/dashboard/homesmetrics/`.
can you make sure existed slugged dashboard url work?

Comment (graceguo-supercat) [reply superset/assets/spec/javascripts/modules/utils_spec.jsx]: i think url is generated by python slugify function...so this JS version is just "tooltip". so it really didn't matter much.

**Review by graceguo-supercat — APPROVED** [review]
LGTM

**Comment by kristw** [comment]
Thank you. Please merge when you are ready.

**Pull request merged**